packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

```diff
@@ -52,8 +47,6 @@ export default util.createRule<Options, MessageIds>({
     /**
      * Sometimes tuple types don't have ObjectFlags.Tuple set, like when they're being matched against an inferred type.
      * So, in addition, check if there are integer properties 0..n and no other numeric keys
-     * @param type type
```

Comment:
Why remove these docs?

Reply:
they have no added value after removing types from jsdoc

Reply:
They still show up in editors when you call the function.

Reply:
This is what that @ param line looks like in vscode
![image](https://user-images.githubusercontent.com/7462525/52494014-b80ac580-2b81-11e9-9c1d-b0513f7a69af.png)

vs no useless comment line:
![image](https://user-images.githubusercontent.com/7462525/52494037-c658e180-2b81-11e9-9468-43c0d3533e4f.png)

I have been going through and removing useless comment lines like this. They add no value to the JSDoc and are ultimately just noise.
Similarly comments like this are just useless noise that add zero value
```TS
/**
 * stringifies a foo
 * @param foo the foo to stringify
 */
function stringifyFoo(foo: Foo): string { /* ... */ }
```